packages/viem-account-ledger/src/ledger-to-account.test.ts

```diff
@@ -37,46 +47,205 @@ describe('ledgerToAccount', () => {
+        for (const v of test_vs) {
+          vi.spyOn(test_ledger, 'signTransaction').mockImplementationOnce(() =>
+            // @ts-expect-error
+            Promise.resolve({
+              v,
+              r: '0x1',
+              s: '0x1',
+            })
+          )
+          await expect(account.signTransaction(txData)).rejects.toThrowError(
+            "Ledger signature `v` was malformed and couldn't be parsed"
+          )
+        }
+      })
     })
   })
 
   it('signs messages', async () => {
-    const account = await ledgerToAccount({
-      transport: await transport,
-    })
     await expect(account.signMessage({ message: 'Hello World' })).resolves.toMatchInlineSnapshot(
       `"0x2f9a547e69592e98114263c08c6f7a6e6cd2f991fc29f442947179419233fe9641c8e4c86975a2722b54313e47768d2ffe2608c497ff9fe7f8c61b12e6257e571c"`
     )
   })
 
   it('signs typed data', async () => {
-    const account = await ledgerToAccount({
+    await expect(
+      account.signTypedData({
+        domain: {
+          name: 'foo',
+          version: '0.0.0',
+          chainId: BigInt(42),
+          verifyingContract: '0x123',
+        },
+        primaryType: 'EIP712Domain',
+        types: {
+          EIP712Domain: [
+            { name: 'name', type: 'string' },
+            { name: 'version', type: 'string' },
+            { name: 'chainId', type: 'uint256' },
+            { name: 'verifyingContract', type: 'address' },
+          ],
+        },
+      })
+    ).rejects.toMatchInlineSnapshot(`[Error: Not implemented as of this release.]`)
+  })
+})
+
+hardwareDescribe('ledgerToAccount (device ledger)', () => {
+  let account: Awaited<ReturnType<typeof ledgerToAccount>>
+  beforeAll(async () => {
+    account = await ledgerToAccount({
       transport: await transport,
     })
+  })
 
+  it('can be setup', async () => {
+    expect((generateLedger as ReturnType<(typeof jest)['fn']>).mock.calls.length).toBe(1)
+  })
+
+  describe('signs txs', () => {
+    const txData = {
+      to: '0x1234567890123456789012345678901234567890',
+      value: BigInt(123),
+      chainId: TEST_CHAIN_ID,
+      nonce: 42,
+      maxFeePerGas: BigInt(100),
+      maxPriorityFeePerGas: BigInt(100),
+    } as const
+
+    describe('eip1559', async () => {
+      test('v=0', async () => {
+        const txHash = await account.signTransaction(txData)
+        const [decoded, signer] = recoverTransaction(txHash)
+        expect(signer.toLowerCase()).toBe(account.address.toLowerCase())
+        // @ts-expect-error
+        expect(decoded.yParity).toMatchInlineSnapshot(`1`)
+      }, 20_000)
+      test('v=1', async () => {
+        const txHash = await account.signTransaction({ ...txData, nonce: 100 })
+        const [decoded, signer] = recoverTransaction(txHash)
+        expect(signer.toLowerCase()).toBe(account.address.toLowerCase())
+        // @ts-expect-error
+        expect(decoded.yParity).toMatchInlineSnapshot(`1`)
```

Comment:
so v=0 is the name here but v=1 is the snap

not a good look

Reply:
maybe dont use snapshot for this since its simple

Reply:
oh oops